### ISSUE TITLE
More precise handling of newlines between methods inside of a type class declaration

### ISSUE THREAD

**Issue by mrkkrp** [tweag/ormolu#110]
Type classes were implemented in #79, but there is still an issue related to white space between methods and data/type family declarations inside of class declarations. I think those should be grouped similar to what we do for top-level functions:

```haskell
class Foo a where
  
  -- | Here goes the foo.
  foo : a -> String
  foo = default_implementation

  -- | Here goes bar.
  bar :: a -> String
  bar = ...
```

Note the newline between `foo` and `bar` and lack of thereof between e.g. type signature of `foo` and its default implementation.

Right now I think everything is joined without newlines, as one block of text.

**Comment by yumiova** [comment]
Indeed, everything is formatted without newlines. I did it that way since no standards had been established yet, and previous Haskell formatters all seemed to use this without-newlines style.

**Comment by mrkkrp** [comment]
@yumiova Feel free to grab this one!

**Comment by yumiova** [comment]
I can implement this change. Should the first line be redundant as shown in the example?

**Comment by yumiova** [comment]
OK, this may take a little bit longer than expected. Somehow, I managed to forget that certain declarations need to be bound together. It's not a particularly difficult task, but finding how to integrate it without completely tearing apart the codebase requires some thought on my part.

**Comment by yumiova** [comment]
I figured out how to implement this without tearing apart the codebase, so this gets implemented in #120.

**Comment by mrkkrp** [comment]
This appears to be resolved now.
